core/sys/windows/psapi.odin

```diff
@@ -0,0 +1,213 @@
+#+build windows
+package sys_windows
+
+import "base:intrinsics"
+import "core:c"
+
+foreign import psapi "system:Psapi.lib"
+
+@(default_calling_convention="system")
+foreign psapi {
+	EnumProcesses :: proc(lpidProcess: PDWORD, cb: DWORD, lpcbNeeded: LPDWORD) -> BOOL ---
+	EnumProcessModules :: proc(hProcess :HANDLE , lphModule: ^HMODULE , cb: DWORD, lpcbNeeded: LPDWORD) -> BOOL ---
+	EnumProcessModulesEx :: proc(hProcess: HANDLE, lphModule: ^HMODULE ,cb: DWORD, lpcbNeeded: LPDWORD , dwFilterFlag: DWORD ) -> BOOL ---
+
+	GetModuleBaseNameW :: proc(hProcess: HANDLE, hModule: HMODULE, lpBaseName: LPWSTR, nSize: DWORD) -> DWORD ---
+	GetModuleFileNameExW :: proc(hProcess: HANDLE, hModule: HMODULE, lpFilename: LPWSTR, nSize: DWORD) -> DWORD ---
+
+	GetModuleInformation :: proc(hProcess: HANDLE, hModule: HMODULE,lpmodinfo: LPMODULEINFO, cb: DWORD) -> BOOL ---
+
+	EmptyWorkingSet :: proc(hProcess: HANDLE) -> BOOL ---
+	QueryWorkingSet :: proc(hProcess: HANDLE,pv: PVOID,cb:DWORD) -> BOOL ---
+	QueryWorkingSetEx :: proc(hProcess: HANDLE, pv: PVOID, cb: DWORD) -> BOOL ---
+
+	InitializeProcessForWsWatch :: proc( hProces: HANDLE) -> BOOL ---
+	GetWsChanges :: proc( hProcess: HANDLE, lpWatchInfo: PPSAPI_WS_WATCH_INFORMATION, cb: DWORD) -> BOOL ---
+	GetWsChangesEx :: proc(hProcess: HANDLE, lpWatchInfoEx: PPSAPI_WS_WATCH_INFORMATION_EX, cb: PDWORD) -> BOOL ---
+
+	GetMappedFileNameW :: proc (hProcess: HANDLE, lpv: LPVOID, lpFilename: LPWSTR, nSize: DWORD) -> DWORD ---
+
+	EnumDeviceDrivers :: proc ( lpImageBase: ^LPVOID, cb: DWORD, lpcbNeeded: LPDWORD) -> BOOL ---
+	GetDeviceDriverBaseNameW :: proc (ImageBase: LPVOID, lpBaseName: LPWSTR, nSize: DWORD) -> DWORD ---
+	GetDeviceDriverFileNameW :: proc (ImageBase: LPVOID, lpFilename: LPWSTR, nSize: DWORD) -> DWORD ---
+
+	GetProcessMemoryInfo :: proc(Process: HANDLE, ppsmemCounters: PPROCESS_MEMORY_COUNTERS, cb: DWORD) -> BOOL ---
+
+	GetPerformanceInfo :: proc( pPerformanceInformation: PPERFORMANCE_INFORMATION, cb: DWORD) -> BOOL ---
+	EnumPageFilesW :: proc(pCallBackRoutine: Enum_Page_File_Callback, pContext: LPVOID) -> BOOL ---
+	GetProcessImageFileNameW :: proc ( hProcess: HANDLE, lpImageFileName: LPWSTR, nSize: DWORD) -> DWORD ---
+}
+
+GetModuleBaseName :: GetModuleBaseNameW
+GetModuleFileNameEx :: GetModuleFileNameExW
+GetMappedFileName :: GetMappedFileNameW
+GetDeviceDriverBaseName :: GetDeviceDriverBaseNameW
+GetDeviceDriverFileName :: GetDeviceDriverFileNameW
+EnumPageFiles :: EnumPageFilesW
+GetProcessImageFileName :: GetProcessImageFileNameW
+
+MODULEINFO :: struct {
+	lpBaseOfDll: LPVOID,
+	SizeOfImage: DWORD,
+	EntryPoint:  LPVOID,
+}
+LPMODULEINFO :: ^MODULEINFO
+
+when ODIN_ARCH == .amd64 { 
+	PSAPI_WORKING_SET_BLOCK :: struct #raw_union {
+		Flags: ULONG_PTR,
+		using DUMMYNAME : bit_field uintptr {
+			Proctection: ULONG_PTR | 5,
+			ShareCount: ULONG_PTR  | 3,
+			Shared :ULONG_PTR | 1,
+			Reserved :ULONG_PTR | 3,
+			VirtualPage: ULONG_PTR | 52,
```

Comment:
Please adhere to the Odin core style guidelines.
```odin
using DUMMYNAME : bit_field u64 {
	Proctection: ULONG_PTR | 5,
	ShareCount:  ULONG_PTR | 3,
	Shared:      ULONG_PTR | 1,
	Reserved:    ULONG_PTR | 3,
	VirtualPage: ULONG_PTR | 52,
}
```

Reply:
This should now be resolved